rewrite-core/src/main/java/org/openrewrite/tree/ParsingExecutionContextView.java

```diff
@@ -15,14 +15,25 @@
  */
 package org.openrewrite.tree;
 
+import org.jetbrains.annotations.NotNull;
 import org.openrewrite.DelegatingExecutionContext;
 import org.openrewrite.ExecutionContext;
+import org.openrewrite.Parser;
+import org.openrewrite.PathUtils;
+import org.openrewrite.internal.StringUtils;
 import org.openrewrite.internal.lang.Nullable;
 
 import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
 
 public class ParsingExecutionContextView extends DelegatingExecutionContext {
     private static final String PARSING_LISTENER = "org.openrewrite.core.parsingListener";
+    private static final String INCLUDE_PATTERNS = "%s.includePatterns";
+    private static final String EXCLUDE_PATTERNS = "%s.excludePatterns";
```

Comment:
I think I now understand that you did it this way so that the `Parser#accept(Path)` can remain as is and that the include and exclude patterns can be used to supply a "delta". Depending on how far we want to take this, that may indeed be a good compromise.